x/evm/precompile/precompile.go

```diff
@@ -203,11 +206,19 @@ func OnRunStart(
 		return res, fmt.Errorf("error committing dirty journal entries: %w", err)
 	}
 
+	// Temporarily switching to a local gas meter to enforce gas limit check for a precompile
+	// returning parent gas meter after execution or failure
+	parentGasMeter := cacheCtx.GasMeter()
+	cacheCtx = cacheCtx.WithGasMeter(storetypes.NewGasMeter(gasLimit)).
+		WithKVGasConfig(storetypes.GasConfig{}).
```

Comment:
Is it okay to pass default `GasConfig`s here?

Reply:
Well the precompile execution calls into regular Cosmos SDK x/module keepers with the provided ctx (cacheCtx in this case), and uses the provided gas config to consume gas. If we pass a default empty `GasConfig` here, then every byte read and written will consume zero gas, so the precompile won't consume any gas at all?